src/jsonpath.spec.ts

```diff
@@ -0,0 +1,656 @@
+
+        expect(parser.getDisclosuresFromJsonpaths(['$.recursive[0]'])).toEqual([mockRecursiveDisclosure.disclosure]);
+
+        expect(parser.getDisclosuresFromJsonpaths(['$.recursive[0].claim[0]'])).toEqual(
+          expect.arrayContaining([mockRecursiveDisclosure.disclosure, mockArrayItem.disclosure]),
+        );
+
+        expect(parser.getDisclosuresFromJsonpaths(['$.recursive[0]', '$.recursive[0].claim[0]'])).toEqual(
+          expect.arrayContaining([mockRecursiveDisclosure.disclosure, mockArrayItem.disclosure]),
+        );
+      });
+
+      it('include recursive parent disclosure', () => {
+        const mockRecursiveDisclosure1: Disclosure = {
+          disclosure: 'mocked-recursive-disclosure',
+          key: 'depth-1',
+          value: {
+            _sd: ['sd-hash-2'],
+          },
+        };
+
+        const mockRecursiveDisclosure2: Disclosure = {
+          disclosure: 'mocked-recursive-disclosure',
+          key: 'depth-2',
+          value: {
+            _sd: ['sd-hash-3'],
+          },
+        };
+
+        const mockRecursiveDisclosure3: Disclosure = {
+          disclosure: 'mocked-recursive-disclosure',
+          key: 'depth-3',
+          value: {
+            _sd: ['sd-hash-4'],
+          },
+        };
+
+        const sdjwt = {
+          _sd: ['sd-hash-1'],
+        };
+
+        const digestMap: DigestMap = new Map([
+          ['sd-hash-1', mockRecursiveDisclosure1],
+          ['sd-hash-2', mockRecursiveDisclosure2],
+          ['sd-hash-3', mockRecursiveDisclosure3],
+          ['sd-hash-4', mockStringDisclosure],
+        ]);
+
+        const parser = new SdJwtDisclosureParser({ jwt: sdjwt, digestMap }).parse();
+
+        expect(parser.getDisclosuresFromJsonpaths(['$.depth-1'])).toEqual([mockRecursiveDisclosure1.disclosure]);
+
+        expect(parser.getDisclosuresFromJsonpaths(['$.depth-1.depth-2'])).toEqual(
+          expect.arrayContaining([mockRecursiveDisclosure1.disclosure, mockRecursiveDisclosure2.disclosure]),
+        );
+
+        expect(parser.getDisclosuresFromJsonpaths(['$.depth-1.depth-2.depth-3'])).toEqual(
+          expect.arrayContaining([
+            mockRecursiveDisclosure1.disclosure,
+            mockRecursiveDisclosure2.disclosure,
+            mockRecursiveDisclosure3.disclosure,
+          ]),
+        );
+
+        expect(parser.getDisclosuresFromJsonpaths(['$.depth-1.depth-2.depth-3.stringClaim'])).toEqual(
+          expect.arrayContaining([
+            mockRecursiveDisclosure1.disclosure,
+            mockRecursiveDisclosure2.disclosure,
+            mockRecursiveDisclosure3.disclosure,
+            mockStringDisclosure.disclosure,
+          ]),
+        );
+      });
+    });
+  });
+});
+
+const hasher = (data) => {
+  const digest = crypto.createHash('sha256').update(data).digest();
+  return base64encode(digest);
+};
+
+const getHasher: GetHasher = () => Promise.resolve(hasher);
+
+describe('#SDJsonpath', () => {
```

Comment:
I would consider moving `SDJsonpath` implementation and tests into separate files.